src/domain/models.py

```
@@ -157,4 +157,110 @@ class FireEvent:
+class LagCorrelation:
+    """Results of lag correlation analysis between stations."""
+    upwind_station_id: int
+    downwind_station_id: int
+    parameter: str
+    lag_hours: int
+    correlation: float
+    p_value: float
+    sample_size: int
+    time_range_start: datetime
+    time_range_end: datetime
+    metadata: Dict[str, Any] = field(default_factory=dict)
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)